lib/gst/player/gstplayer.c

```diff
@@ -1660,6 +2229,204 @@ gst_player_get_pipeline (GstPlayer * self)
   return val;
 }
 
+/**
+ * gst_player_get_media_info:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): Current media info instance.
+ * the caller should free it with g_object_unref()
+ */
+GstPlayerMediaInfo*
+gst_player_get_media_info (GstPlayer *self)
+{
+  GstPlayerMediaInfo *info;
+
+  g_mutex_lock (&self->priv->lock);
+  info = gst_player_media_info_copy (self->priv->media_info);
+  g_mutex_unlock (&self->priv->lock);
+
+  return info;
+}
+
+/**
+ * gst_player_get_current_audio_track:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): current audio track
+ * the caller should free it with g_object_unref()
+ */
+GstPlayerAudioInfo*
+gst_player_get_current_audio_track (GstPlayer *self)
+{
+  GstPlayerAudioInfo *info;
+
+  if (!is_track_enabled (self, GST_PLAY_FLAG_AUDIO))
+    return NULL;
+
+  info = (GstPlayerAudioInfo*) gst_player_stream_info_get_current (self,
+            "current-audio", GST_TYPE_PLAYER_AUDIO_INFO);
+  return info;
+}
+
+/**
+ * gst_player_get_current_video_track:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer full): current video track
+ * the caller should free it with g_object_unref()
+ */
+GstPlayerVideoInfo*
+gst_player_get_current_video_track (GstPlayer *self)
+{
+  GstPlayerVideoInfo *info;
+
+  if (!is_track_enabled (self, GST_PLAY_FLAG_VIDEO))
+    return NULL;
+
+  info = (GstPlayerVideoInfo*) gst_player_stream_info_get_current (self,
+            "current-video", GST_TYPE_PLAYER_VIDEO_INFO);
+  return info;
+}
+
+/**
+ * gst_player_get_current_subtitle_track:
+ * @player: #GstPlayer instance
+ *
+ * Returns: (transfer none): current subtitle track
+ */
+GstPlayerSubtitleInfo*
+gst_player_get_current_subtitle_track (GstPlayer *self)
+{
+  GstPlayerSubtitleInfo *info;
+
+  if (!is_track_enabled (self, GST_PLAY_FLAG_SUBTITLE))
+    return NULL;
+
+  info = (GstPlayerSubtitleInfo*) gst_player_stream_info_get_current (self,
+            "current-text", GST_TYPE_PLAYER_SUBTITLE_INFO);
+  return info;
+}
+
+/**
+ * gst_player_set_audio_track:
+ * @player: #GstPlayer instance
+ * @stream_index: stream index
+ */
+gboolean
+gst_player_set_audio_track (GstPlayer *self, gint stream_index)
+{
+  GstPlayerStreamInfo *info;
+
+  g_mutex_lock (&self->priv->lock);
+  info = gst_player_stream_info_find (self, self->priv->media_info,
+            GST_TYPE_PLAYER_AUDIO_INFO, stream_index);
+  g_mutex_unlock (&self->priv->lock);
+  if (!info) {
+    GST_ERROR_OBJECT (self, "invalid audio stream index %d", stream_index);
+    return FALSE;
+  }
+
```

Comment:
just curious, why do I need to unref() info ? The gst_player_stream_info_find() will return the location of the streamInfo in our streams_list. We don't call g_object_ref() inside the stream_find() function to increase the ref count. The function is an internal function. the goal of function is so check if given stream_index exist in our stream_list.